### PR TITLE
Add missing aiofiles.tempfile classes

### PR DESCRIPTION
**Pull request by andrewshadura**
I was trying to add missing aiofiles.tempfile classes, but I ran into this issue:

```
error: not checking stubs due to mypy build errors:
stubs/aiofiles/aiofiles/tempfile/__init__.pyi:132: error: Overloaded function signatures 1 and 2 overlap with incompatible return types
stubs/aiofiles/aiofiles/tempfile/__init__.pyi:132: error: Overloaded function signatures 1 and 3 overlap with incompatible return types
stubs/aiofiles/aiofiles/tempfile/__init__.pyi:132: error: Overloaded function signatures 1 and 4 overlap with incompatible return types
stubs/aiofiles/aiofiles/tempfile/__init__.pyi:146: error: Overloaded function signatures 2 and 3 overlap with incompatible return types
stubs/aiofiles/aiofiles/tempfile/__init__.pyi:146: error: Overloaded function signatures 2 and 4 overlap with incompatible return types
```

I’m not sure how to proceed from here.

This will fix: #6524

### CODE REVIEW

File: stubs/aiofiles/aiofiles/tempfile/__init__.pyi [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+    loop: AbstractEventLoop | None = ...,
+    executor: Any | None = ...,
+) -> AiofilesContextManager[None, None, AsyncBufferedIOBase]: ...
+def TemporaryDirectory(
+    suffix: StrOrBytesPath | None = ...,
+    prefix: StrOrBytesPath | None = ...,
+    dir: StrOrBytesPath | None = ...,
+    loop: AbstractEventLoop | None = ...,
+    executor: Any | None = ...,
+) -> StrOrBytesPath: ...
```
Comment (srittau) [inline stubs/aiofiles/aiofiles/tempfile/__init__.pyi:198]: This return an `AiofilesContextManagerTempDir`.

**Review by srittau — COMMENTED** [review]
The warnings can be suppresed with `# type: ignore[misc]` on the affected lines. I noticed one more thing for now.

**Comment by AlexWaygood** [comment]
There's a lot of pyright errors due to various parameters not being annotated. Ideally we'd obviously annotate these, obviously, but for now we could also maybe just add specific files to the allowlist in `pyrightconfig.stricter.json`:

https://github.com/python/typeshed/blob/4e946b74bf18dc932caf41d0257957d85bbabe91/pyrightconfig.stricter.json#L9-L76

File: stubs/aiofiles/aiofiles/tempfile/__init__.pyi [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+def TemporaryDirectory(
+    suffix: bytes | None = ...,
+    prefix: bytes | None = ...,
+    dir: BytesPath | None = ...,
+    loop: AbstractEventLoop | None = ...,
+    executor: Incomplete | None = ...,
+) -> AiofilesContextManagerTempDir[None, None, AsyncTemporaryDirectory]: ...
+
+class AiofilesContextManagerTempDir(AiofilesContextManager[_T_co, _T_contra, _V_co]):
+    async def __aenter__(self): ...
```
Comment (AlexWaygood) [inline stubs/aiofiles/aiofiles/tempfile/__init__.pyi:260]:
```suggestion
    async def __aenter__(self) -> Incomplete: ...
```

File: stubs/aiofiles/aiofiles/tempfile/__init__.pyi [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+@overload
+def TemporaryDirectory(
+    suffix: bytes | None = ...,
+    prefix: bytes | None = ...,
+    dir: BytesPath | None = ...,
+    loop: AbstractEventLoop | None = ...,
+    executor: Any | None = ...,
+) -> AiofilesContextManagerTempDir[None, None, AsyncTemporaryDirectory]: ...
+
+class AiofilesContextManagerTempDir(AiofilesContextManager[_T_co, _T_contra, _V_co]): ...
```
[the remark this answers is not in the document]
Comment (AlexWaygood) [reply stubs/aiofiles/aiofiles/tempfile/__init__.pyi]: I think it returns `str`, but I'm not 100% sure, so I've gone with `Incomplete` for now.

**Review by srittau — APPROVED** [review]

**Pull request merged**